FirebaseInstallations/Source/Library/Public/FIRInstallations.h

```diff
@@ -14,39 +14,101 @@
+#import <FirebaseInstallations/FIRInstallationsAuthTokenResult.h>
+
 @class FIRApp;
 
 NS_ASSUME_NONNULL_BEGIN
 
-extern NSString *const FIRInstallationIDDidChangeNotification;
+/** A notification with this name is sent each time an installation is created or deleted. */
+FOUNDATION_EXPORT const NSNotificationName FIRInstallationIDDidChangeNotification;
 
+/**
+ * An installation ID handler block.
+ * @param identifier The instalation ID string if exists or @c nil otherwise.
+ * @param error The error when @c identifier==nil or @c nil otherwise.
+ */
 typedef void (^FIRInstallationsIDHandler)(NSString *__nullable identifier,
                                           NSError *__nullable error)
     NS_SWIFT_NAME(InstallationsIDHandler);
 
+/**
+ * An authorization token handler block.
+ * @param tokenResult An instance of @c FIRInstallationsAuthTokenResult in case of success or @c nil
+ * otherwise.
+ * @param error The error when @c tokenResult==nil or @c nil otherwise.
+ */
 typedef void (^FIRInstallationsTokenHandler)(
     FIRInstallationsAuthTokenResult *__nullable tokenResult, NSError *__nullable error)
     NS_SWIFT_NAME(InstallationsTokenHandler);
 
+/**
+ * The class provides API for Firebase Installations.
+ * Each configured @c FIRApp has a corresponding single instance of @c FIRInstallations.
+ * An instance of the class provides access to the installation info for the @c FIRApp as well as
+ * allows to delete it. A Firebase Installation is unique by @c FIRApp.name and @c
+ * FIRApp.options.googleAppID .
+ */
 NS_SWIFT_NAME(Installations)
 @interface FIRInstallations : NSObject
 
-// TODO: Add `+ (FIRInstallations *)installations;`
+/**
+ * Returns a default instance of @c FIRInstallations.
+ * @return Returns an instance of @c FIRInstallations for @c[FIRApp defaultApp]. Returns @c nil if
+ * the default app is not configured yet.
+ */
++ (nullable FIRInstallations *)installations;
 
+/**
+ * Returns an instance of @c FIRInstallations for an application.
+ * @param application A configured @c FIRApp instance.
+ * @return Returns an instance of @c FIRInstallations corresponding to the passed application.
+ */
 + (FIRInstallations *)installationsWithApp:(FIRApp *)application NS_SWIFT_NAME(installations(app:));
 
+/**
+ * The method creates or retrieves an installation ID. The installation ID is a stable identifier
+ * that uniquely identifies the app instance. NOTE: If the application already has an existing
+ * FirebaseInstanceID then the InstanceID ideintifier will be used.
+ * @param completion A completion handler which is invoked when the operation completes. See @c
+ * FIRInstallationsIDHandler for additional details.
+ */
 - (void)installationIDWithCompletion:(FIRInstallationsIDHandler)completion;
 
+/**
+ * Retrives (locally if exists or from the server) a valid authorization token. An existing token
+ * may be invalidated or expire, so it is recommended the auth token before each server request. The
```

Comment:
"invalidated or expired".

What's recommended with the auth token? To refresh it?